teamcity: failed test: TestInitialPartitioning

The following tests appear to have failed on master (testrace): TestInitialPartitioning/group: TestInitialPartitioning/group/0: TestInitialPartitioning/group/0/multi_col_range_partitioning_-_MAXVALUE, TestInitialPartitioning/group: TestInitialPartitioning/group/0: TestInitialPartitioning/group/0/multi_col_list_partitioning_-_DEFAULT_DEFAULT_subpartitioned, TestInitialPartitioning/group: TestInitialPartitioning/group/0: TestInitialPartitioning/group/0/list-list_partitioning, TestInitialPartitioning/group: TestInitialPartitioning/group/0, TestInitialPartitioning, TestInitialPartitioning/group

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestInitialPartitioning).

[#1343788](https://teamcity.cockroachdb.com/viewLog.html?buildId=1343788):


```
TestInitialPartitioning
--- FAIL: testrace/TestInitialPartitioning (884.200s)




TestInitialPartitioning/group: TestInitialPartitioning/group/0
--- FAIL: test/TestInitialPartitioning: TestInitialPartitioning/group: TestInitialPartitioning/group/0 (890.930s)

------- Stdout: -------
=== PAUSE TestInitialPartitioning/group/0
------- Stdout: -------
=== PAUSE TestInitialPartitioning/group/0



TestInitialPartitioning/group: TestInitialPartitioning/group/0: TestInitialPartitioning/group/0/list-list_partitioning
...root] will distribute plan: true
                      [n1,client=127.0.0.1:60404,user=root] execution starts: distributed engine
                       === SPAN START: consuming rows ===
                      [n1,client=127.0.0.1:60404,user=root] creating DistSQL plan with isLocal=false
                      [n1,client=127.0.0.1:60404,user=root] querying next range at /Table/71/1/5/7
                      [n1,client=127.0.0.1:60404,user=root] running DistSQL plan
                       === SPAN START: flow ===
                      [n1,client=127.0.0.1:60404,user=root] starting (0 processors, 0 startables)
                       === SPAN START: table reader ===
                    cockroach.processorid: 0
                    cockroach.stat.tablereader.bytes.read: 0 B
                    cockroach.stat.tablereader.input.rows: 0
                    cockroach.stat.tablereader.stalltime: 2µs
                      [n1,client=127.0.0.1:60404,user=root] starting scan with limitBatches true
                      [n1,client=127.0.0.1:60404,user=root] Scan /Table/71/1/{5/7-6}
                       === SPAN START: txn coordinator send ===
                       === SPAN START: dist sender send ===
                      [n1,client=127.0.0.1:60404,user=root,txn=37cc98f3] querying next range at /Table/71/1/5/7
                      [n1,client=127.0.0.1:60404,user=root,txn=37cc98f3] r161: sending batch 1 Scan to (n1,s1):1
                      [n1,client=127.0.0.1:60404,user=root,txn=37cc98f3] sending request to local client
                       === SPAN START: /cockroach.roachpb.Internal/Batch ===
                      [n1] 1 Scan
                      [n1,s1] executing 1 requests
                      [n1,s1,r161/1:/Table/71/1/{5/7-6}] read-only path
                      [n1,s1,r161/1:/Table/71/1/{5/7-6}] read has no clock uncertainty
                      [n1,s1,r161/1:/Table/71/1/{5/7-6}] acquire latches
                      [n1,s1,r161/1:/Table/71/1/{5/7-6}] waited 93.199µs to acquire latches
                      [n1,s1,r161/1:/Table/71/1/{5/7-6}] waiting for read lock
                      [n1,s1,r161/1:/Table/71/1/{5/7-6}] read completed
                       === SPAN START: count rows ===
                    cockroach.processorid: 1
                    cockroach.stat.aggregator.input.rows: 0
                    cockroach.stat.aggregator.mem.max: 0 B
                    cockroach.stat.aggregator.stalltime: 624µs
                      [n1,client=127.0.0.1:60404,user=root] execution ends
                      [n1,client=127.0.0.1:60404,user=root] rows affected: 1
                      [n1,client=127.0.0.1:60404,user=root] AutoCommit. err: <nil>
                      [n1,client=127.0.0.1:60404,user=root] releasing 1 tables
                       === SPAN START: exec cmd: exec stmt ===
                      [n1,client=127.0.0.1:60404,user=root] [NoTxn pos:8482] executing ExecStmt: SET TRACING = off
                      [n1,client=127.0.0.1:60404,user=root] executing: SET TRACING = off in state: NoTxn
                    goroutine 1085786 [running]:
                    runtime/debug.Stack(0xa7a358200, 0xc009f92360, 0x506dec0)
                    	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
                    github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x51365c0, 0xc001b29600, 0xc009f92330)
                    	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:47 +0x181
                    github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestInitialPartitioning.func1.1.1(0xc001b29600)
                    	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:1201 +0x22d
                    testing.tRunner(0xc001b29600, 0xc0067d8e40)
                    	/usr/local/go/src/testing/testing.go:865 +0x164
                    created by testing.(*T).Run
                    	/usr/local/go/src/testing/testing.go:916 +0x65b



TestInitialPartitioning/group
--- FAIL: testrace/TestInitialPartitioning: TestInitialPartitioning/group (0.000s)




TestInitialPartitioning/group: TestInitialPartitioning/group/0: TestInitialPartitioning/group/0/multi_col_range_partitioning_-_MAXVALUE
...ot] will distribute plan: true
                      [n1,client=127.0.0.1:60404,user=root] execution starts: distributed engine
                       === SPAN START: consuming rows ===
                      [n1,client=127.0.0.1:60404,user=root] creating DistSQL plan with isLocal=false
                      [n1,client=127.0.0.1:60404,user=root] querying next range at /Table/69/1/5/6
                      [n1,client=127.0.0.1:60404,user=root] running DistSQL plan
                       === SPAN START: flow ===
                      [n1,client=127.0.0.1:60404,user=root] starting (0 processors, 0 startables)
                       === SPAN START: table reader ===
                    cockroach.processorid: 0
                    cockroach.stat.tablereader.bytes.read: 0 B
                    cockroach.stat.tablereader.input.rows: 0
                    cockroach.stat.tablereader.stalltime: 2µs
                      [n1,client=127.0.0.1:60404,user=root] starting scan with limitBatches false
                      [n1,client=127.0.0.1:60404,user=root] Scan /Table/69/1/5/6{-/#}
                       === SPAN START: txn coordinator send ===
                       === SPAN START: dist sender send ===
                      [n1,client=127.0.0.1:60404,user=root,txn=f883ea47] querying next range at /Table/69/1/5/6
                      [n1,client=127.0.0.1:60404,user=root,txn=f883ea47] r143: sending batch 1 Scan to (n1,s1):1
                      [n1,client=127.0.0.1:60404,user=root,txn=f883ea47] sending request to local client
                       === SPAN START: /cockroach.roachpb.Internal/Batch ===
                      [n1] 1 Scan
                      [n1,s1] executing 1 requests
                      [n1,s1,r143/1:/Table/69/1/5/{6-7}] read-only path
                      [n1,s1,r143/1:/Table/69/1/5/{6-7}] read has no clock uncertainty
                      [n1,s1,r143/1:/Table/69/1/5/{6-7}] acquire latches
                      [n1,s1,r143/1:/Table/69/1/5/{6-7}] waited 144.674µs to acquire latches
                      [n1,s1,r143/1:/Table/69/1/5/{6-7}] waiting for read lock
                      [n1,s1,r143/1:/Table/69/1/5/{6-7}] read completed
                       === SPAN START: count rows ===
                    cockroach.processorid: 1
                    cockroach.stat.aggregator.input.rows: 0
                    cockroach.stat.aggregator.mem.max: 0 B
                    cockroach.stat.aggregator.stalltime: 506µs
                      [n1,client=127.0.0.1:60404,user=root] execution ends
                      [n1,client=127.0.0.1:60404,user=root] rows affected: 1
                      [n1,client=127.0.0.1:60404,user=root] AutoCommit. err: <nil>
                      [n1,client=127.0.0.1:60404,user=root] releasing 1 tables
                       === SPAN START: exec cmd: exec stmt ===
                      [n1,client=127.0.0.1:60404,user=root] [NoTxn pos:7384] executing ExecStmt: SET TRACING = off
                      [n1,client=127.0.0.1:60404,user=root] executing: SET TRACING = off in state: NoTxn
                    goroutine 936030 [running]:
                    runtime/debug.Stack(0xa7a358200, 0xc00addfec0, 0x506dec0)
                    	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
                    github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x51365c0, 0xc001a46900, 0xc00addfe90)
                    	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:47 +0x181
                    github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestInitialPartitioning.func1.1.1(0xc001a46900)
                    	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:1201 +0x22d
                    testing.tRunner(0xc001a46900, 0xc0072f34a0)
                    	/usr/local/go/src/testing/testing.go:865 +0x164
                    created by testing.(*T).Run
                    	/usr/local/go/src/testing/testing.go:916 +0x65b



TestInitialPartitioning/group: TestInitialPartitioning/group/0
--- FAIL: testrace/TestInitialPartitioning: TestInitialPartitioning/group: TestInitialPartitioning/group/0 (884.190s)

------- Stdout: -------
=== PAUSE TestInitialPartitioning/group/0



TestInitialPartitioning
--- FAIL: test/TestInitialPartitioning (899.900s)
None



TestInitialPartitioning/group
--- FAIL: test/TestInitialPartitioning: TestInitialPartitioning/group (0.000s)
None



TestInitialPartitioning/group: TestInitialPartitioning/group/0: TestInitialPartitioning/group/0/multi_col_list_partitioning_-_DEFAULT_DEFAULT_subpartitioned
...ent=127.0.0.1:60404,user=root] will distribute plan: true
                      [n1,client=127.0.0.1:60404,user=root] execution starts: distributed engine
                       === SPAN START: consuming rows ===
                      [n1,client=127.0.0.1:60404,user=root] creating DistSQL plan with isLocal=false
                      [n1,client=127.0.0.1:60404,user=root] querying next range at /Table/61/1/6
                      [n1,client=127.0.0.1:60404,user=root] running DistSQL plan
                       === SPAN START: flow ===
                      [n1,client=127.0.0.1:60404,user=root] starting (0 processors, 0 startables)
                       === SPAN START: table reader ===
                    cockroach.processorid: 0
                    cockroach.stat.tablereader.bytes.read: 0 B
                    cockroach.stat.tablereader.input.rows: 0
                    cockroach.stat.tablereader.stalltime: 2µs
                      [n1,client=127.0.0.1:60404,user=root] starting scan with limitBatches true
                      [n1,client=127.0.0.1:60404,user=root] Scan /Table/61/{1/6-2}
                       === SPAN START: txn coordinator send ===
                       === SPAN START: dist sender send ===
                      [n1,client=127.0.0.1:60404,user=root,txn=3492427d] querying next range at /Table/61/1/6
                      [n1,client=127.0.0.1:60404,user=root,txn=3492427d] r97: sending batch 1 Scan to (n1,s1):1
                      [n1,client=127.0.0.1:60404,user=root,txn=3492427d] sending request to local client
                       === SPAN START: /cockroach.roachpb.Internal/Batch ===
                      [n1] 1 Scan
                      [n1,s1] executing 1 requests
                      [n1,s1,r97/1:/Table/61/{1/6-2}] read-only path
                      [n1,s1,r97/1:/Table/61/{1/6-2}] read has no clock uncertainty
                      [n1,s1,r97/1:/Table/61/{1/6-2}] acquire latches
                      [n1,s1,r97/1:/Table/61/{1/6-2}] waited 138.225µs to acquire latches
                      [n1,s1,r97/1:/Table/61/{1/6-2}] waiting for read lock
                      [n1,s1,r97/1:/Table/61/{1/6-2}] read completed
                       === SPAN START: count rows ===
                    cockroach.processorid: 1
                    cockroach.stat.aggregator.input.rows: 0
                    cockroach.stat.aggregator.mem.max: 0 B
                    cockroach.stat.aggregator.stalltime: 389µs
                      [n1,client=127.0.0.1:60404,user=root] execution ends
                      [n1,client=127.0.0.1:60404,user=root] rows affected: 1
                      [n1,client=127.0.0.1:60404,user=root] AutoCommit. err: <nil>
                      [n1,client=127.0.0.1:60404,user=root] releasing 1 tables
                       === SPAN START: exec cmd: exec stmt ===
                      [n1,client=127.0.0.1:60404,user=root] [NoTxn pos:3787] executing ExecStmt: SET TRACING = off
                      [n1,client=127.0.0.1:60404,user=root] executing: SET TRACING = off in state: NoTxn
                    goroutine 500662 [running]:
                    runtime/debug.Stack(0xa7a358200, 0xc006dbc8d0, 0x506dec0)
                    	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
                    github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x51365c0, 0xc00597df00, 0xc006dbc840)
                    	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:47 +0x181
                    github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestInitialPartitioning.func1.1.1(0xc00597df00)
                    	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:1201 +0x22d
                    testing.tRunner(0xc00597df00, 0xc00981c480)
                    	/usr/local/go/src/testing/testing.go:865 +0x164
                    created by testing.(*T).Run
                    	/usr/local/go/src/testing/testing.go:916 +0x65b




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/ccl/partitionccl/partition_test.go
+++ b/pkg/ccl/partitionccl/partition_test.go
@@ -1161,50 +1161,26 @@ func TestInitialPartitioning(t *testing.T) {
 	}
 
 	rng, _ := randutil.NewPseudoRand()
-	allTestCases := allPartitioningTests(rng)
+	testCases := allPartitioningTests(rng)
 
-	parallelism := 8
-	if util.RaceEnabled {
-		// Reduce parallelism for race tests, or we can generate too many
-		// goroutines.
-		parallelism = 2
-	}
-	splitTestCases := make([][]partitioningTest, parallelism)
-	for i := range splitTestCases {
-		n := len(allTestCases) / (parallelism - i)
-		splitTestCases[i] = allTestCases[:n]
-		allTestCases = allTestCases[n:]
-	}
+	ctx := context.Background()
+	db, sqlDB, cleanup := setupPartitioningTestCluster(ctx, t)
+	defer cleanup()
 
-	// Group together parallel subtests, so that the main test doesn't run its
-	// defer before the parallel subtests start.
-	t.Run("group", func(t *testing.T) {
-		for instanceIdx := range splitTestCases {
-			testCases := splitTestCases[instanceIdx]
-			t.Run(fmt.Sprintf("%d", instanceIdx), func(t *testing.T) {
-				t.Parallel() // SAFE FOR TESTING
-				ctx := context.Background()
-				db, sqlDB, cleanup := setupPartitioningTestCluster(ctx, t)
-				defer cleanup()
-
-				for _, test := range testCases {
-					if len(test.scans) == 0 {
-						continue
-					}
-					t.Run(test.name, func(t *testing.T) {
-						if err := test.parse(); err != nil {
-							t.Fatalf("%+v", err)
-						}
-						sqlDB.Exec(t, test.parsed.createStmt)
-						sqlDB.Exec(t, test.parsed.zoneConfigStmts)
-
-						testutils.SucceedsSoon(t, test.verifyScansFn(ctx, db))
-					})
-				}
-			})
+	for _, test := range testCases {
+		if len(test.scans) == 0 {
+			continue
 		}
+		t.Run(test.name, func(t *testing.T) {
+			if err := test.parse(); err != nil {
+				t.Fatalf("%+v", err)
+			}
+			sqlDB.Exec(t, test.parsed.createStmt)
+			sqlDB.Exec(t, test.parsed.zoneConfigStmts)
 
-	})
+			testutils.SucceedsSoon(t, test.verifyScansFn(ctx, db))
+		})
+	}
 }
 
 func TestSelectPartitionExprs(t *testing.T) {
@@ -1291,115 +1267,96 @@ func TestRepartitioning(t *testing.T) {
 		t.Skip()
 	}
 
-	runTest := func(t *testing.T, db *gosql.DB, test repartitioningTest) {
-		sqlDB := sqlutils.MakeSQLRunner(db)
-		sqlDB.Exec(t, `DROP DATABASE IF EXISTS data`)
-		sqlDB.Exec(t, `CREATE DATABASE data`)
-
-		if err := test.old.parse(); err != nil {
-			t.Fatalf("%+v", err)
-		}
-		sqlDB.Exec(t, test.old.parsed.createStmt)
-		sqlDB.Exec(t, test.old.parsed.zoneConfigStmts)
-
-		testutils.SucceedsSoon(t, test.old.verifyScansFn(context.Background(), db))
-
-		if err := test.new.parse(); err != nil {
-			t.Fatalf("%+v", err)
-		}
-		sqlDB.Exec(t, fmt.Sprintf("ALTER TABLE %s RENAME TO %s", test.old.parsed.tableName, test.new.parsed.tableName))
-
-		testIndex, _, err := test.new.parsed.tableDesc.FindIndexByName(test.index)
-		if err != nil {
-			t.Fatalf("%+v", err)
-		}
-
-		var repartition bytes.Buffer
-		if testIndex.ID == test.new.parsed.tableDesc.PrimaryIndex.ID {
-			fmt.Fprintf(&repartition, `ALTER TABLE %s `, test.new.parsed.tableName)
-		} else {
-			fmt.Fprintf(&repartition, `ALTER INDEX %s@%s `, test.new.parsed.tableName, testIndex.Name)
-		}
-		if testIndex.Partitioning.NumColumns == 0 {
-			repartition.WriteString(`PARTITION BY NOTHING`)
-		} else {
-			if err := sql.ShowCreatePartitioning(
-				&sqlbase.DatumAlloc{}, test.new.parsed.tableDesc, testIndex,
-				&testIndex.Partitioning, &repartition, 0 /* indent */, 0, /* colOffset */
-			); err != nil {
-				t.Fatalf("%+v", err)
-			}
-		}
-		sqlDB.Exec(t, repartition.String())
-
-		// Verify that repartitioning removes zone configs for partitions that
-		// have been removed.
-		newPartitionNames := map[string]struct{}{}
-		for _, name := range test.new.parsed.tableDesc.PartitionNames() {
-			newPartitionNames[name] = struct{}{}
-		}
-		rows := sqlDB.QueryStr(t, "SELECT cli_specifier FROM [SHOW ALL ZONE CONFIGURATIONS] WHERE cli_specifier IS NOT NULL")
-		for _, row := range rows {
-			zs, err := config.ParseCLIZoneSpecifier(row[0])
-			if err != nil {
-				t.Fatal(err)
-			}
-			if !zs.TargetsTable() {
-				// Ignore zone configs that target databases or system ranges.
-				continue
-			}
-			if zs.TableOrIndex.Table.Table() != test.new.parsed.tableDesc.Name || zs.Partition == "" {
-				// Ignore zone configs that do not target a partition of this table.
-				continue
-			}
-			if _, ok := newPartitionNames[string(zs.Partition)]; !ok {
-				t.Errorf("zone config for removed partition %q exists after repartitioning", zs.Partition)
-			}
-		}
-
-		// NB: Not all old zone configurations are removed. This statement will
-		// overwrite any with the same name and the repartitioning removes any
-		// for partitions that no longer exist, but there could still be some
-		// sitting around (e.g., when a repartitioning preserves a partition but
-		// does not apply a new zone config). This is fine.
-		sqlDB.Exec(t, test.new.parsed.zoneConfigStmts)
-
-		testutils.SucceedsSoon(t, test.new.verifyScansFn(context.Background(), db))
-	}
-
 	rng, _ := randutil.NewPseudoRand()
-	allTestCases, err := allRepartitioningTests(allPartitioningTests(rng))
+	testCases, err := allRepartitioningTests(allPartitioningTests(rng))
 	if err != nil {
 		t.Fatalf("%+v", err)
 	}
-	parallelism := 8
-	if util.RaceEnabled {
-		// Reduce parallelism for race tests, or we can generate too many
-		// goroutines.
-		parallelism = 2
-	}
-	splitTestCases := make([][]repartitioningTest, parallelism)
-	for i := range splitTestCases {
-		n := len(allTestCases) / (parallelism - i)
-		splitTestCases[i] = allTestCases[:n]
-		allTestCases = allTestCases[n:]
-	}
 
-	// Group together parallel subtests, so that the main test doesn't run its
-	// defer before the parallel subtests start.
-	t.Run("group", func(t *testing.T) {
-		for instanceIdx := range splitTestCases {
-			testCases := splitTestCases[instanceIdx]
-			t.Run(fmt.Sprintf("%d", instanceIdx), func(t *testing.T) {
-				t.Parallel() // SAFE FOR TESTING
-				db, _, cleanup := setupPartitioningTestCluster(context.Background(), t)
-				defer cleanup()
-				for _, test := range testCases {
-					runTest(t, db, test)
+	ctx := context.Background()
+	db, sqlDB, cleanup := setupPartitioningTestCluster(ctx, t)
+	defer cleanup()
+
+	for _, test := range testCases {
+		t.Run(fmt.Sprintf("%s/%s", test.old.name, test.new.name), func(t *testing.T) {
+			sqlDB.Exec(t, `DROP DATABASE IF EXISTS data`)
+			sqlDB.Exec(t, `CREATE DATABASE data`)
+
+			{
+				if err := test.old.parse(); err != nil {
+					t.Fatalf("%+v", err)
 				}
-			})
-		}
-	})
+				sqlDB.Exec(t, test.old.parsed.createStmt)
+				sqlDB.Exec(t, test.old.parsed.zoneConfigStmts)
+
+				testutils.SucceedsSoon(t, test.old.verifyScansFn(ctx, db))
+			}
+
+			{
+				if err := test.new.parse(); err != nil {
+					t.Fatalf("%+v", err)
+				}
+				sqlDB.Exec(t, fmt.Sprintf("ALTER TABLE %s RENAME TO %s", test.old.parsed.tableName, test.new.parsed.tableName))
+
+				testIndex, _, err := test.new.parsed.tableDesc.FindIndexByName(test.index)
+				if err != nil {
+					t.Fatalf("%+v", err)
+				}
+
+				var repartition bytes.Buffer
+				if testIndex.ID == test.new.parsed.tableDesc.PrimaryIndex.ID {
+					fmt.Fprintf(&repartition, `ALTER TABLE %s `, test.new.parsed.tableName)
+				} else {
+					fmt.Fprintf(&repartition, `ALTER INDEX %s@%s `, test.new.parsed.tableName, testIndex.Name)
+				}
+				if testIndex.Partitioning.NumColumns == 0 {
+					repartition.WriteString(`PARTITION BY NOTHING`)
+				} else {
+					if err := sql.ShowCreatePartitioning(
+						&sqlbase.DatumAlloc{}, test.new.parsed.tableDesc, testIndex,
+						&testIndex.Partitioning, &repartition, 0 /* indent */, 0, /* colOffset */
+					); err != nil {
+						t.Fatalf("%+v", err)
+					}
+				}
+				sqlDB.Exec(t, repartition.String())
+
+				// Verify that repartitioning removes zone configs for partitions that
+				// have been removed.
+				newPartitionNames := map[string]struct{}{}
+				for _, name := range test.new.parsed.tableDesc.PartitionNames() {
+					newPartitionNames[name] = struct{}{}
+				}
+				rows := sqlDB.QueryStr(t, "SELECT cli_specifier FROM [SHOW ALL ZONE CONFIGURATIONS] WHERE cli_specifier IS NOT NULL")
+				for _, row := range rows {
+					zs, err := config.ParseCLIZoneSpecifier(row[0])
+					if err != nil {
+						t.Fatal(err)
+					}
+					if !zs.TargetsTable() {
+						// Ignore zone configs that target databases or system ranges.
+						continue
+					}
+					if zs.TableOrIndex.Table.Table() != test.new.parsed.tableDesc.Name || zs.Partition == "" {
+						// Ignore zone configs that do not target a partition of this table.
+						continue
+					}
+					if _, ok := newPartitionNames[string(zs.Partition)]; !ok {
+						t.Errorf("zone config for removed partition %q exists after repartitioning", zs.Partition)
+					}
+				}
+
+				// NB: Not all old zone configurations are removed. This statement will
+				// overwrite any with the same name and the repartitioning removes any
+				// for partitions that no longer exist, but there could still be some
+				// sitting around (e.g., when a repartitioning preserves a partition but
+				// does not apply a new zone config). This is fine.
+				sqlDB.Exec(t, test.new.parsed.zoneConfigStmts)
+
+				testutils.SucceedsSoon(t, test.new.verifyScansFn(ctx, db))
+			}
+		})
+	}
 }
 
 func TestRemovePartitioningExpiredLicense(t *testing.T) {

